### DISCUSSION TITLE
dbeaver hangs since latest upgrade

### DISCUSSION THREAD

**Discussion by BobSmiley** [dbeaver/dbeaver#21464] [General]
I received a notification about an update being available and installed it, and since then when launching dbeaver the main console loads but immediately becomes unresponsive with the windows "not responding".  I let it run for about an hour just in case it would sort itself out, but in the end I had to kill it.  I've relaunched and rekilled the process multiple times, so I think this is just my "new normal".  My toolbar says dbeaver 23.2.1, but I can't get to the "about" menu to get more specific than that.

I've seen the recommendation to reset UI settings, but I can't get to the help menu to do that.

I don't see any widespread alerts about an issue like this with the last release, so I suspect there's an issue on my side, does anyone have any suggestions?

Thanks in advance.

**Comment by E1izabeth** [comment]
Please, provide the log files. You can find instructions here https://dbeaver.com/docs/wiki/Log-files/.

**Reply by BobSmiley** [reply]
Thanks Elizabeth.  Here are both logs mentioned in the link you sent.
[dbeaver-debug.log](https://github.com/dbeaver/dbeaver/files/12872512/dbeaver-debug.log)
[Copy.log](https://github.com/dbeaver/dbeaver/files/12872515/Copy.log)

(The Copy.log file is the .log file that I added "Copy" to just to make it attachable to this message.)

**Reply by E1izabeth** [reply]
Thanks. Unfortunately I don't see something that may lead to hangs in the logs. Do you see some progress bar with a message at the right bottom corner when DBeaver is hanging?

**Comment by E1izabeth** [comment]
Also what OS do you use?

**Reply by BobSmiley** [reply]
Thanks Elizabeth.  I'm using Windows 11.  (No smirks please...it's a corporate requirement...)  :-)

**Comment by BobSmiley** [comment]
In my attempts today, I find that (a) there was a new version, so I downloaded it and installed it, and (b) now when I'm connecting to a database, I get prompted with an internal dbeaver browser window for the MFA portion of the login...previously this opened in a tab in my system browser.  The unfortunate part of this new way of doing things with the internal browser is that the credentials aren't cached, and so I'm repeatedly prompted for my password before I can even open a single database, and each time I provide my password, I'm sent a text from Microsoft to authenticate.  Typically I get a single text every 2 weeks and I enter a password once or twice per day; now I'm having to enter it dozens of times per day with no apparent progress.  I bet this is at the root of my issue.

Is there perhaps a new system setting I can use to revert the configuration to use the system browser instead of the internal dbeaver browser?

BTW, my databases are hosted on Azure and I'm using Azure MFA authentication.

**Reply by E1izabeth** [reply]
Could you try switching this setting?
![image](https://github.com/dbeaver/dbeaver/assets/28875055/0df97c23-606f-4477-8248-b9a8ad160ada)

It should help.

**Reply by BobSmiley** [reply]
THAT DID IT!  Thanks so much.
I skipped right over that settings page because it looked like a folder for the sub-pages; didn't occur to me that there would be actual settings _on_ that page!  This corrected my whole issue.

Thanks again @E1izabeth. I really appreciate your helping me on this.

**Comment by BobSmiley** [comment]
BTW, I just found the "Web Browser" preference in the User Interface section, and it's set to "Use external web browser" and "default system web browser".  So I've set this to "Edge" to see if that will make a difference.  I thought perhaps the "default system web browser" wasn't working well so it reverted to the internal browser despite the setting.  But setting this configuration and then restarting dbeaver didn't change the behavior.

**Reply by Destrolaric** [reply]
@BobSmiley Sorry to bother you, but do you have problems connecting to SQL Server? Because currently, SQL Server driver has a bug when in tries to call SSO requests multiple times if there are any issues with connection.

**Comment by BobSmiley** [comment]
@Destrolaric 
 Hi Giorgii, since reconfiguring with help from this forum I no longer have an issue connecting to sql server on azure using MFA. Before this reconfiguration, I did see the issue you speak about. The only way to resolve it was to kill the app.

Thanks for the follow-up!

Bob Smiley
(Sent from my iPhone mail)

On Oct 23, 2023, at 6:11 AM, Georgii ***@***.***> wrote:

﻿

@BobSmiley<https://github.com/BobSmiley> Sorry to bother you, but do you have problems connecting to SQL Server? Because currently, SQL Server driver has a bug when in tries to call SSO requests multiple times if there are any issues with connection.
